littleblog/entry/serializers.py

```diff
@@ -1,15 +1,36 @@
 from rest_framework import serializers
 from .models import Entry, UserProfile
 from django.contrib.auth.models import User
+from rest_framework.validators import UniqueValidator
+from django.contrib.auth.password_validation import UserAttributeSimilarityValidator, MinimumLengthValidator, \
+    CommonPasswordValidator, NumericPasswordValidator
 
 
 class UserProfileCreateSerializer(serializers.ModelSerializer):
 
-    user = serializers.PrimaryKeyRelatedField(queryset=User.objects.all())
+    UserProfile = serializers.ReadOnlyField(source='profile.nickname')
+    nickname = serializers.CharField(max_length=50, required=True, write_only=True)
+    avatar = serializers.ImageField(required=False, write_only=True)
 
     class Meta:
-        model = UserProfile
-        fields = '__all__'
+        model = User
+        fields = ('id', 'username', 'password', 'nickname', 'avatar', 'UserProfile')
+        extra_kwargs = {
+            'password': {'write_only': True, 'validators': [CommonPasswordValidator().validate,
+                                                            MinimumLengthValidator().validate,
+                                                            NumericPasswordValidator().validate,
+                                                            UserAttributeSimilarityValidator().validate]
```

Comment:
I don't think you need to specify all of those.
Did you test it without this? I think DRF/Django should run these validations by default and you only should specify validators in settings.AUTH_PASSWORD_VALIDATORS

Reply:
My goal is to make validation at the serializer.is_valid stage, otherwise I could meet a problem while attempting to create DB entry with improper values (using user.objects.create_user in serializer.create method)

Reply:
Ok, seems like that's a proper way, but you don't need to specify all validators here - you have them on settings, so you can reuse django utils for passwords.


```
from django.contrib.auth.password_validation import validate_password

class Serializer ...
...
def validate_password(self, value):
    validate_password(value)
    return value
```
Let me know if this doesn't work - we will figure it out